### DISCUSSION TITLE
Accessing class attribute from classmethod

### DISCUSSION THREAD

**Discussion by Fkawala** [pydantic/pydantic#2894] [Question] [answered]
Hello,

I'd like to access the class attribute from a class method. Here is a minimum reproducible example.

```python
import pydantic

class X(pydantic.BaseModel):
    a_class_attribute: str = "a default value"

    @classmethod
    def a_class_method(cls):
        print(cls.a_class_attribute)

X.a_class_method()

"""
Traceback (most recent call last):
  File "file.py", line 12, in <module>
    X.a_class_method()
  File "file.py", line 10, in a_class_method
    print(cls.a_class_attribute)
AttributeError: type object 'X' has no attribute 'a_class_attribute'
"""
```
For what it worth, this is valid python:
```python
class X():
    a_class_attribute: str = "a default value"

    @classmethod
    def a_class_method(cls):
        print(cls.a_class_attribute)

X.a_class_method()

"""
a default value
"""
```

What would be the solution here ? I'm quite sure that I'm missing something obvious. 

Thank you !

PS. There is [another discussion](https://github.com/samuelcolvin/pydantic/discussions/2854) that (I believe) shares the same root cause, but I think it worth to have this in a different discussion with a simpler/more isolated example.

**Comment by PrettyWood** [comment, marked as answer]
Hi @Fkawala 
Change `str` into `ClassVar[str]` ;)

**Reply by Fkawala** [reply]
Works like a charm, thanks a lot for the blazing fast answer! From the [doc](https://pydantic-docs.helpmanual.io/usage/models/#automatically-excluded-attributes) I understand that using `ClassVar` excludes my class attribute from the model. Is that correct?
